sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/ColumnType.scala

```diff
@@ -705,6 +705,37 @@ private[columnar] case class MAP(dataType: MapType)
   override def clone(v: UnsafeMapData): UnsafeMapData = v.copy()
 }
 
+private[columnar] object CALENDAR_INTERVAL extends ColumnType[CalendarInterval]
+  with DirectCopyColumnType[CalendarInterval] {
+
+  override def dataType: DataType = CalendarIntervalType
+
+  override def defaultSize: Int = 16
+
+  override def actualSize(row: InternalRow, ordinal: Int): Int = 20
+
+  override def getField(row: InternalRow, ordinal: Int): CalendarInterval = row.getInterval(ordinal)
+
+  override def setField(row: InternalRow, ordinal: Int, value: CalendarInterval): Unit = {
+    row.setInterval(ordinal, value)
+  }
+
+  override def extract(buffer: ByteBuffer): CalendarInterval = {
+    ByteBufferHelper.getInt(buffer)
```

Comment:
the first int is the size?

Reply:
Now I got why the actual size is 4 bytes larget than default size. We should document it.

BTW can we skip storing the size? It's fixed anyway.

Reply:
yes, we can skip it, the current implementation is just to reuse copy logic of `MutableUnsafeRow`